app/components/provider_interface/filter_checkbox_component.rb

```diff
@@ -0,0 +1,16 @@
+module ProviderInterface
+  class FilterCheckboxComponent < ActionView::Component::Base
+    include ViewHelper
+
+    attr_reader :name, :text, :value, :selected, :heading
+
+    def initialize(name:, text:, value:, selected:, heading:)
+      @name = name
```

Comment:
It could in this case but I wanted to make it generic as possible - I could foresee a future scenario whereby you want different names and values and therefore have left them as separate.

Reply:
Had a conversation with Duncan about this will look to simplify it when we move to integrate with controllers etc (see comment below for more context).